### PR TITLE
Copyq improvements

### PR DESCRIPTION
**Pull request by Tyilo**
For the last commit ("Properly handle escaping of underlined texts"), the problem before can be seen by copying `<foo> "bar"` and then searching for the empty string.

### CODE REVIEW

File: CopyQ.py [hunk 1]
```diff
@@ -69,7 +69,8 @@ def handleQuery(query):
                 )
             ]
 
-        script = copyq_script_getMatches % query.string if query.string else copyq_script_getAll
+        escaped_query = query.string.replace('\\', '\\\\').replace('"', '\\"')
```
Comment (ManuelSchneid3r) [inline CopyQ.py]: why is this necessary?

File: CopyQ.py [hunk 2]
```diff
@@ -55,21 +55,48 @@
...
             text = json_obj['text']
             if not text:
                 text = "<i>No text</i>"
             else:
-                text = pattern.sub(lambda m: "<u>%s</u>" % m.group(0), html.escape(" ".join(filter(None, text.replace("\n", " ").split(" ")))))
+                U_BEGIN = "@U_BEGIN_XXXXX@"
+                U_END = "@U_END_XXXXX@"
+                raw_text = " ".join(filter(None, text.replace("\n", " ").split(" ")))
+                underlined_text = pattern.sub(lambda m: U_BEGIN + m.group(0) + U_END, raw_text)
+                text = html.escape(underlined_text).replace(U_BEGIN, "<u>").replace(U_END, "</u>")
```
Comment (ManuelSchneid3r) [inline CopyQ.py]: why this indirection over U_BEGIN and U_END?

File: CopyQ.py (hunk 1 above)
Comment (Tyilo) [reply CopyQ.py]: Otherwise the `script` will not have valid js syntax if the query includes `"` or `\\`.

File: CopyQ.py (hunk 2 above)
Comment (Tyilo) [reply CopyQ.py]: The problem is that if you have copied `<u></u>` (and other tags), we want to escape them so we don't view them raw.
We still want to add `<u></u>` to the parts of the string that matched the query, so we need this indirection.

**Comment by ManuelSchneid3r** [comment]
Damn I just went through it to notice that I screwed up by moving them to dirs. If you could fix that I'll merge instantly

**Comment by ManuelSchneid3r** [comment]
tely I let you all wait pretty long. Now I have the problem that I don't know which PR's are valid at all. Therefore I will close all PR's for tentatively. If you want me to reopen the PR let me know. Also it would be nice if you could resolve potential conflicts due to changes in the repo. Sorry for the inconveniences, but I am working on it for a week now. I'm in a hurry. Note that this is a copy paste message. 😉
